internal/system/snapd_client.go

```diff
@@ -0,0 +1,155 @@
+					return (&net.Dialer{
+						Timeout:   30 * time.Second,
+						KeepAlive: 30 * time.Second,
+					}).DialContext(ctx, "unix", "/run/snapd.socket")
+				},
+			},
+			Timeout: 60 * time.Second,
+		},
+	}
+}
+
+// SnapdResponse represents the common structure of snapd API responses.
+// See https://snapcraft.io/docs/using-the-api
+type SnapdResponse struct {
+	Type       string          `json:"type"`
+	StatusCode int             `json:"status-code"`
+	Status     string          `json:"status"`
+	Result     json.RawMessage `json:"result"`
+}
+
+// snapdSnap represents information about a snap from the snapd API.
+// See https://snapcraft.io/docs/snapd-rest-api#heading--snaps
+type snapdSnap struct {
+	ID              string                 `json:"id"`
+	Name            string                 `json:"name"`
+	Status          string                 `json:"status"`
+	Version         string                 `json:"version"`
+	Revision        string                 `json:"revision"`
+	Channel         string                 `json:"channel"`
+	TrackingChannel string                 `json:"tracking-channel"`
+	Confinement     string                 `json:"confinement"`
+	Channels        map[string]ChannelInfo `json:"channels"`
+}
+
+// ChannelInfo represents channel-specific information for a snap.
+type ChannelInfo struct {
+	Revision    string `json:"revision"`
+	Confinement string `json:"confinement"`
+	Version     string `json:"version"`
+	Channel     string `json:"channel"`
+}
+
+// GetSnap queries information about an installed snap.
+func (c *SnapdClient) GetSnap(name string) (*snapdSnap, *SnapdResponse, error) {
+	apiURL := fmt.Sprintf("http://localhost/v2/snaps/%s", url.PathEscape(name))
+
+	req, err := http.NewRequest("GET", apiURL, nil)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to execute request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	var snapdResp SnapdResponse
+	if err := json.Unmarshal(body, &snapdResp); err != nil {
+		return nil, nil, fmt.Errorf("failed to unmarshal response: %w", err)
+	}
+
+	if snapdResp.StatusCode == 404 {
+		return nil, &snapdResp, fmt.Errorf("snap not installed: %s", name)
+	}
+	if snapdResp.StatusCode != 200 {
+		return nil, &snapdResp, fmt.Errorf("unexpected status code: %d", snapdResp.StatusCode)
+	}
+
+	var snapInfo snapdSnap
+	if err := json.Unmarshal(snapdResp.Result, &snapInfo); err != nil {
+		return nil, &snapdResp, fmt.Errorf("failed to unmarshal snap info: %w", err)
+	}
+
+	return &snapInfo, &snapdResp, nil
+}
+
+// FindOne searches for a snap in the snap store.
+func (c *SnapdClient) FindOne(name string) (*snapdSnap, *SnapdResponse, error) {
+	apiURL := fmt.Sprintf("http://localhost/v2/find?name=%s", url.QueryEscape(name))
```

Comment:
I think this is fine with `url.QueryEscape`, but it might be slightly more native to do it with `url.Values`:

```go
query := url.Values{"name": []string{name}}
apiURL := "http://localhost/v2/find?" + query.Encode()
```